pkg/controller/gc_controller.go

```diff
@@ -130,12 +137,43 @@ func (c *gcController) processQueueItem(key string) error {
 		return nil
 	}
 
-	log.Info("Backup has expired. Creating a DeleteBackupRequest.")
+	log.Info("Backup has expired")
 
-	req := pkgbackup.NewDeleteBackupRequest(backup.Name, string(backup.UID))
+	selector := labels.SelectorFromSet(labels.Set(map[string]string{
+		arkv1api.BackupNameLabel: backup.Name,
+		arkv1api.BackupUIDLabel:  string(backup.UID),
+	}))
 
-	_, err = c.deleteBackupRequestClient.DeleteBackupRequests(ns).Create(req)
+	dbrs, err := c.deleteBackupRequestLister.DeleteBackupRequests(ns).List(selector)
 	if err != nil {
+		return errors.Wrap(err, "error listing existing DeleteBackupRequests for backup")
+	}
+
+	// if there's an existing unprocessed deletion request for this backup, don't create
+	// another one
+	for _, dbr := range dbrs {
+		switch dbr.Status.Phase {
+		case "", arkv1api.DeleteBackupRequestPhaseNew, arkv1api.DeleteBackupRequestPhaseInProgress:
+			log.Info("Backup already has a pending deletion request")
+			return nil
+		}
+	}
+
+	// otherwise, any deletion requests that exist have already been processed (and must have
+	// failed), so delete them all
```

Comment:
I have a feeling it might be better to handle this in the backup deletion controller - we need to cap the manual creation of DBRs at 1 also. WDYT?

Reply:
"this" being just the part that deletes existing processed backups?  Yeah, that seems reasonable to move over.  I wasn't too worried about manual DBRs but may as well handle them the same.

I'd leave the section that checks for an existing unprocessed request, and doesn't create a new one if there's already one, in the GC controller - that make sense?

Reply:
> "this" being just the part that deletes existing processed backups? Yeah, that seems reasonable to move over. I wasn't too worried about manual DBRs but may as well handle them the same.

Correct. I'm thinking in the backup deletion controller, when we process a DBR, we list all the DBRs for the associated backup, and delete all except the one being processed.

> I'd leave the section that checks for an existing unprocessed request, and doesn't create a new one if there's already one, in the GC controller - that make sense? Obv putting it as-is in the DBR controller wouldn't work.

Agreed, leave this in GC

Reply:
I guess the other question is, if deleting existing DBRs in the DBR controller, what should we do with new/in-progress DBRs for the same backup?  I suppose we could just delete them - given that we're currently only running one worker, anything with a phase of In Progress that's not the current DBR isn't *actually* being processed

Reply:
Let's imagine we run multiple workers. Workers 1 and 2 each get a DBR for the same backup (2 different DBRs). There are a couple of ways this could play out:

1. Worker 1 gets DBR 1
1. Worker 1 lists all DBRs for the backup and sees 0 InProgress
1. Worker 1 sets status to InProgress
1. Worker 2 gets DBR 2
1. Worker 2 lists all DBRs for the backup and sees 1 InProgress
1. Worker 2 deletes DBR 2 because there's another DBR In Progress

or

1. Worker 1 gets DBR 1
1. Worker 2 gets DBR 2
1. Worker 1 lists all DBRs and sees none InProgress so it sets DBR 1 to InProgress
1. Worker 2 lists all DBRs and sees none InProgress (informer cache hasn't seen DBR 1 InProgress update yet) so it sets DBR 2 to InProgress
1. Now we have 2 workers trying to delete the same backup :-(

In both scenarios, we could instead use a shared "in progress DBRs" data structure to ensure there's never more than 1 InProgress at a time.